META.md

```diff
@@ -96,3 +108,135 @@ meta exec "git commit -s -m 'Removed integtest.sh.'"
 meta exec "git push username remove-integtest-sh"
 meta exec "gh pr create --title 'Removing default integtest.sh.' --body='Coming from https://github.com/opensearch-project/opensearch-build/issues/497, removing default integtest.sh.'"
 ```
+
+### Increment a Version in Every Plugin
+
+Because one cannot install an older version of a plugin on a newer version of OpenSearch (see [OpenSearch#1707](https://github.com/opensearch-project/OpenSearch/issues/1707)), it's common to have to increment versions in all plugins, without making other changes. This was the case in [the 1.2.3 release](https://github.com/opensearch-project/opensearch-build/issues/1365).
+
+See also [opensearch-build#1375](https://github.com/opensearch-project/opensearch-build/issues/1375) which aims to supersede this labor-intensive process with plugins auto-incrementing versions for the next development iteration.
+
+#### Increment Version in OpenSearch
+
+Increment the version in OpenSearch patch branch, e.g. 1.2, [OpenSearch#1758](https://github.com/opensearch-project/OpenSearch/pull/1758). After this change is merged, backport the version increment change in OpenSearch to `1.x` ([OpenSearch#1759](https://github.com/opensearch-project/OpenSearch/pull/1759)) and `main` ([OpenSearch#1760](https://github.com/opensearch-project/OpenSearch/pull/1760)).
+
+#### Create a 1.2.3 Manifest
+
+Create a new manifest that only contains OpenSearch, e.g. [opensearch-build#1369](https://github.com/opensearch-project/opensearch-build/pull/1369). After this manifest is merged, wait for a successful SNAPSHOT build.
+
+#### Increment Version in Plugins
+
+Check out and update the 1.2 branch.
+
+```
+meta git update
+meta git pull origin
+meta git checkout 1.2
+meta git pull origin 1.2
+```
+
+Replace known versions.
+
+```
+find . -name build.gradle -print0 | xargs -0 sed -i "s/1.2.2-SNAPSHOT/1.2.3-SNAPSHOT/g"
+find . -wholename "*/.github/workflows/*.yml" -print0 | xargs -0 sed -i "s/1.2.2-SNAPSHOT/1.2.3-SNAPSHOT/g"
+find . -wholename "*/.github/workflows/*.yml" -print0 | xargs -0 sed -i "s/1.2.2.0-SNAPSHOT/1.2.3.0-SNAPSHOT/g"
+```
+
+Plugins such as k-nn and security need some exact version replacements.
+
+```
+find . -name build.gradle -print0 | xargs -0 sed -i "s/1.2.2/1.2.3/g"
+find . -name CMakeLists.txt -print0 | xargs -0 sed -i "s/1.2.2.0/1.2.3.0/g"
+find . -name pom.xml -print0 | xargs -0 sed -i "s/1.2.2/1.2.3/g"
```

Comment:
Beware of blanked find/replace statements. In security, for example, bumping from 1.2.3 to 1.2.4 would have also matched this https://github.com/opensearch-project/security/blob/main/pom.xml#L81

Reply:
Fair enough! I think it’s a poor man’s implementation anyway. What we really need is GHA workflows in each repo that increment versions based on created tags.

Reply:
Here are two attempts at that:

https://github.com/opensearch-project/OpenSearch/pull/1816
https://github.com/opensearch-project/common-utils/pull/106